File: movie-recommendation-backend-new/src/main/java/com/movierecommendation/backend/model/data/WatchProviders.java
package com.movierecommendation.backend.model.data;

import lombok.Data;

import java.util.List;

@Data
public class WatchProviders {
    private String link;
    private List<Provider> flatrate;
    private List<Provider> rent;
    private List<Provider> buy;

    @Data
    public static class Provider {
        private String logo_path;
        private int provider_id;
        private String provider_name;
        private int display_priority;
    }
}

File: movie-recommendation-frontend/src/models/WatchProviders.ts
export class Provider {
  logo_path: string;
  provider_id: number;
  provider_name: string;
  display_priority: number;

  constructor() {
    this.logo_path = '';
    this.provider_id = 0;
    this.provider_name = '';
    this.display_priority = 0;
  }
}

export class WatchProviders {
  link: string;
  flatrate: Provider[];
  rent: Provider[];
  buy: Provider[];

  constructor() {
    this.link = '';
    this.flatrate = [];
    this.rent = [];
    this.buy = [];
  }
}
